fix(ivy): handle namespaces in attributes

crisbeto opened the pull request:
Adds handling for namespaced attributes when generating the template and in the `elementAttribute` instruction.

This PR resolves FW-672.

kara reviewed (approved):
LGTM

kara commented:
[presubmit)[http://test/OCL:230380276:BASE:230380379:1548186003269:17873a4a]

alxhub commented on packages/compiler/src/render3/view/template.ts:
```diff
@@ -719,13 +721,22 @@ export class TemplateDefinitionBuilder implements t.Visitor<void>, LocalResolver
+
+            // If there wasn't a sanitization ref, we need to add
+            // an extra param so that we can pass in the namespace.
+            sanitizationRef ? params.push(namespaceLiteral) :
```
Please use a real `if` statement - having side effects hidden behind a ternary is not very readable.

alxhub commented on packages/compiler/src/render3/view/template.ts:
```diff
@@ -1261,6 +1270,25 @@ function getLiteralFactory(
+ * Gets an array of literals that can be added to an expression
+ * to represent the name and namespace of an attribute. E.g.
+ * `:xlink:href` turns into `[AttributeMarker.NamespaceURI, 'xlink', 'href']`.
+ * @param name Name of the attribute, including the namespace.
```
Nit: blank line between the comment and the `@param`

crisbeto commented:
I've addressed the feedback @alxhub.

crisbeto committed:
fix(ivy): handle namespaces in attributes

Adds handling for namespaced attributes when generating the template and in the `elementAttribute` instruction.

JoostK commented on packages/compiler/src/render3/view/template.ts:
```diff
@@ -719,13 +721,25 @@ export class TemplateDefinitionBuilder implements t.Visitor<void>, LocalResolver
+          if (attrNamespace) {
+            const namespaceLiteral = o.literal(attrNamespace);
+
+            if (sanitizationRef) {
```
why not

```typescript
if (!sanitizationRef) {
  // If there wasn't a sanitization ref, we need to add
  // an extra param so that we can pass in the namespace.
  params.push(o.literal(null));
}

params.push(namespaceLiteral);
```

kara commented:
merge assistance: looks like the "size" job is stuck again
